### PR TITLE
New docs

### PR DESCRIPTION
**Pull request by scart97**
* Finish Ultimate guide
* Add new doc about implementing the data pipeline

### CODE REVIEW

File: docs/Custom Data.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+4. `self.open_text` uses the same metadata to open the corresponding text label
+5. `self.preprocess_text` can be used to apply any transform directly to the text. Common options are lower case, expanding contractions (`I'm` becomes `I am`), expanding numbers (`42` becomes `forty two`) and removing punctuation
+
+## Example: Loading data from nemo
+
+This example will implement `thunder.data.datamodule.ManifestDatamodule` and `thunder.data.dataset.ManifestSpeechDataset`.
+
+### Load source
+
+The nemo manifest file contains one json in each line, with the relevant data:
```
Comment (rbracco) [inline docs/Custom Data.md:43]: Because it's a bit confusing how NeMo chose to do this, I might elaborate a bit and say something like:

Although NeMo manifest files have a .json extension, they are not actually legal json. They instead follow the [JSON Lines](https://jsonlines.org/) standard where each line is a complete JSON object representing a single audio and its metadata.

File: docs/Ultimate guide.md [hunk 2]
```diff
@@ -1,12 +1,12 @@
-# The ultimate guide to speech recognition (WIP)
+# The ultimate guide to speech recognition
 
 This guide has the purpose to give you all the steps necessary to achieve a decent (but not necessarily state-of-the-art) speech recognition system in a new language.
```
Comment (rbracco) [inline docs/Ultimate guide.md:3]: "This guide is meant to give you..." reads more naturally

File: docs/Ultimate guide.md [hunk 3]
```diff
@@ -66,18 +66,55 @@ Try to fix those label problems, or remove them from the training set if you hav
 Don't spend weeks just looking at the data, but have a small subset that you can trust is properly cleaned, even if that means manually labeling again.
 After you train the first couple of models, it's possible to use the model itself to help find problems in the training data.
 
-## Writing the dataset/datamodule
+## Loading the data
 
-`TODO: fill this section with the nemo manifest example`
+The first step to train a speech recognition model is to load the collected data into the specific format the model expects.
+Usually, data from different sources will have unique ways in that the label is encoded.
```
Comment (rbracco) [inline docs/Ultimate guide.md:72]: Reads better if you remove the word "in" from this line

File: docs/Ultimate guide.md [hunk 4]
```diff
@@ -66,18 +66,55 @@ Try to fix those label problems, or remove them from the training set if you hav
...
+{"audio_filepath": "commonvoice/pt/train/20272843.mp3", "duration": 2.544, "text": "Eu vou desligar"}
+```
+
+These three keys for each example are required:
+
+* **audio_filepath**: Contains the path to the input audio
+* **duration**: has the duration of the audio, in seconds
+* **text**: that's the corresponding label
+
+Make sure that each example starts and end in the same line.
```
Comment (rbracco) [inline docs/Ultimate guide.md:100]: "in" should be "on"

File: docs/Ultimate guide.md [hunk 5]
```diff
@@ -100,15 +137,28 @@ the final value will be lower. This means that small bumps will happen, but it w
 and keep going down. The ideal point is where you run the prediction on the batch that you overfit,
```
Comment (rbracco) [inline docs/Ultimate guide.md:137]: It seems like some of the text got cut off here

File: docs/Ultimate guide.md [hunk 6]
```diff
@@ -100,15 +137,28 @@ the final value will be lower. This means that small bumps will happen, but it w
...
+![train_loss](images/first_train/train_loss.png)
+
+The validation loss is not important at this stage, we are trying to overfit on purpose to check if the model is learning correctly.
+It's possible to notice that it improved in the first epoch, but instantly went back up and keep increasing as the training goes.
+
+Expected validation loss:
+
+![val_loss](images/first_train/val_loss.png)
+
 Some problems that can happen:
```
Comment (rbracco) [inline docs/Ultimate guide.md:152]: This section is awesome and will be extremely helpful for people newer to ASR
